src/stores/datStore.ts

```diff
@@ -194,6 +283,62 @@ export default class DatStore {
         return event;
     }
 
+    async fetchBuyReturn(
+        datAddress: string,
+        totalPaid: BigNumber
+    ): Promise<BuyReturnCached> {
+        const blockNumber = this.rootStore.providerStore.getCurrentBlockNumber();
+
+        const tokensIssued = await this.estimateBuyValue(datAddress, totalPaid);
+        const pricePerToken = totalPaid.div(tokensIssued);
+
+        return {
+            value: {
+                totalPaid,
+                tokensIssued,
+                pricePerToken,
+            },
+            blockNumber,
+        };
+    }
+
+    async fetchSellReturn(
+        datAddress: string,
+        tokensSold: BigNumber
+    ): Promise<SellReturnCached> {
+        const blockNumber = this.rootStore.providerStore.getCurrentBlockNumber();
+
+        const currencyReturned = await this.estimateSellValue(
+            datAddress,
+            tokensSold
+        );
+
+        const returnPerToken = currencyReturned.div(tokensSold);
+
+        return {
+            value: {
+                tokensSold,
+                currencyReturned,
+                returnPerToken,
+            },
+            blockNumber,
+        };
+    }
+
+    async fetchSpotPrice(datAddress: string): Promise<BigNumber> {
```

Comment:
We're not using this now, but this would be used with the chart right?

Reply:
Yup